internal/ls/utilities.go

```diff
@@ -724,3 +1525,106 @@ func getAllSuperTypeNodes(node *ast.Node) []*ast.TypeNode {
+		func(t *checker.Type) *ast.Symbol {
+			return core.IfElse(t.Symbol() != nil && t.Symbol().Flags&(ast.SymbolFlagsClass|ast.SymbolFlagsInterface) != 0, t.Symbol(), nil)
+		},
+	)
+	if len(res) == 0 {
+		return nil
+	}
+	return res
+}
+
+/**
+* Find symbol of the given property-name and add the symbol to the given result array
+* @param symbol a symbol to start searching for the given propertyName
+* @param propertyName a name of property to search for
+* @param result an array of symbol of found property symbols
+* @param previousIterationSymbolsCache a cache of symbol from previous iterations of calling this function to prevent infinite revisiting of the same symbol.
+*                                The value of previousIterationSymbol is undefined when the function is first called.
+ */
+func getPropertySymbolsFromBaseTypes(symbol *ast.Symbol, propertyName string, checker *checker.Checker, cb func(base *ast.Symbol) *ast.Symbol) *ast.Symbol {
+	seen := core.Set[*ast.Symbol]{}
+	var recur func(*ast.Symbol) *ast.Symbol
+	recur = func(symbol *ast.Symbol) *ast.Symbol {
+		// Use `addToSeen` to ensure we don't infinitely recurse in this situation:
+		//      interface C extends C {
+		//          /*findRef*/propName: string;
+		//      }
+		if symbol.Flags&(ast.SymbolFlagsClass|ast.SymbolFlagsInterface) == 0 || !seen.AddIfAbsent(symbol) {
+			return nil
+		}
+
+		return core.FirstNonNil(symbol.Declarations, func(declaration *ast.Declaration) *ast.Symbol {
+			return core.FirstNonNil(getAllSuperTypeNodes(declaration), func(typeReference *ast.TypeNode) *ast.Symbol {
+				propertyType := checker.GetTypeAtLocation(typeReference)
+				if propertyType == nil || propertyType.Symbol() == nil {
+					return nil
+				}
+				propertySymbol := checker.GetPropertyOfType(propertyType, propertyName)
+				// Visit the typeReference as well to see if it directly or indirectly uses that property
+				if propertySymbol != nil {
+					if r := core.FirstNonNil(checker.GetRootSymbols(propertySymbol), cb); r != nil {
+						return r
+					}
+				}
+				return recur(propertyType.Symbol())
+			})
+		})
+	}
+	return recur(symbol)
+}
+
+func getPropertySymbolFromBindingElement(checker *checker.Checker, bindingElement *ast.Node) *ast.Symbol {
+	if typeOfPattern := checker.GetTypeAtLocation(bindingElement.Parent); typeOfPattern != nil {
+		return checker.GetPropertyOfType(typeOfPattern, bindingElement.Name().Text())
+	}
+	return nil
+}
+
+func getPropertySymbolOfObjectBindingPatternWithoutPropertyName(symbol *ast.Symbol, checker *checker.Checker) *ast.Symbol {
+	bindingElement := ast.GetDeclarationOfKind(symbol, ast.KindBindingElement)
+	if bindingElement != nil && isObjectBindingElementWithoutPropertyName(bindingElement) {
+		return getPropertySymbolFromBindingElement(checker, bindingElement)
+	}
+	return nil
+}
+
+func getSourceOfDefaultedAssignment(node *ast.Node) *ast.Node {
```

Comment:
not needed because this is called from unneeded code.
also, defaulted assignments (`window.x = window.x || "thing"`) are not supported in Corsa